### PR TITLE
separate endpoints for accounts, appointments, businesses, expenses, services, and staff

### PR DESCRIPTION
**Pull request by gateremark**

### CODE REVIEW

**Comment by gateremark** [comment]
Have successfully separated the first endpoints in utils directory to their separate files and created a PR but after separating the other API endpoints to separate files, the deployment has failed, possibly because of change on where the API endpoints are being hit.

File: app/api/accounts.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { useAppDispatch } from "@/hooks";
+import { setMessage } from "@/store/toastSlice";
+import { apiCall } from "@/utils/apiRequest";
+import endpoints from "@/utils/endpoints";
+import { useMutation, useQuery } from "react-query";
+
+// Create Expense Accounts
+export const useCreateExpenseAccounts = () => {
+  const dispatch = useAppDispatch();
+  return useMutation<void, Error, any>(async (accounts) => {
```
Comment (daveclinton) [inline app/api/accounts.ts:10]: Add appropriate type

File: app/api/accounts.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    try {
+      const response = await apiCall(
+        "POST",
+        endpoints.createAccount,
+        accounts,
+        {}
+      );
+      dispatch(setMessage(response.message));
```
Comment (daveclinton) [inline app/api/accounts.ts:18]: Check for undefined

File: app/api/businesses.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import { CloudinaryData } from "@/components/types";
+import { useAppDispatch } from "@/hooks";
+import { setMessage } from "@/store/toastSlice";
+import { apiCall } from "@/utils/apiRequest";
+import endpoints from "@/utils/endpoints";
+import { useMutation, useQuery } from "react-query";
+
+export const useAssignService = () => {
+  const dispatch = useAppDispatch();
+  return useMutation<void, Error, any>(async (services: any) => {
```
Comment (daveclinton) [inline app/api/businesses.ts:10]: Add appropriate type

File: app/api/businesses.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      );
+      return response || {};
+    } catch (error) {
+      throw new Error("Error fetching Statuses");
+    }
+  });
+};
+
+export const useGetAllServices = () => {
+  return useQuery("", async () => {
```
Comment (daveclinton) [inline app/api/businesses.ts:64]: Add response key

File: app/api/expenses.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      return response;
+    } catch (error) {
+      throw new Error("Error fetching Expenses");
+    }
+  });
+};
+
+export const useCreateExpense = () => {
+  const dispatch = useAppDispatch();
+  return useMutation<void, Error, any>(
```
Comment (daveclinton) [inline app/api/expenses.ts:20]: Add appropriate type

**Review by daveclinton — CHANGES_REQUESTED** [review]

File: app/api/accounts.ts (hunk 1 above)
Comment (gateremark) [reply app/api/accounts.ts]: Done

File: app/api/accounts.ts (hunk 2 above)
Comment (gateremark) [reply app/api/accounts.ts]: Done

File: app/api/businesses.ts (hunk 3 above)
Comment (gateremark) [reply app/api/businesses.ts]: Done

File: app/api/expenses.ts (hunk 5 above)
Comment (gateremark) [reply app/api/expenses.ts]: Done

**Review by daveclinton — CHANGES_REQUESTED** [review]
@gateremark Change Base branch to release

**Comment by gateremark** [comment]
@daveclinton sorry, I didn't get what you mean by change the Base branch to release

**Comment by gateremark** [comment]
@daveclinton, has led to conflicts and some mess. Trying to resolve might take some time.

**Comment by gateremark** [comment]
Created other Pull Requests that are better organized with no conflicts and with release as the base branch

**Pull request closed without merging**